Add race page

SaamIAm opened the pull request:
**Basic Race page:**
includes canvas with spammy race feature of block moving across screen. 
Feel free to roast, I was not sure why Canvas is above the banner. 
This is a WIP open to tips and tricks.

**Must Haves:**
-needs win condition (when block crosses threshold) ✅ 
-needs timer/timing feature (race time) ✅ 
-needs score feature (turns race time into score) ✅ 
-needs API connection
-needs endurance feature (button cool down) ✅

**Could Haves:**
-could use restart/reset feature ✅
-could use transition smoothing between race jumps (jerky but could be aesthetic)
-could use art and better styling

SaamIAm committed:
linked barebones Race page to index

SaamIAm committed:
linked barebones Race page to index

anishprasad01 committed:
Created StatusRequest type
Created PetStatus type
Added new endpoint to Pet (DOES NOT WORK YET)

anishprasad01 committed:
Created StatusRequest type
Created PetStatus type
Added new endpoint to Pet (DOES NOT WORK YET)

anishprasad01 committed:
Created StatusRequest type
Created PetStatus type
Added new endpoint to Pet (DOES NOT WORK YET)

anishprasad01 committed:
Created StatusRequest type
Created PetStatus type
Added new endpoint to Pet (DOES NOT WORK YET)

anishprasad01 committed:
Makes GetPetStatus temp compile

anishprasad01 committed:
Makes GetPetStatus temp compile

anishprasad01 committed:
Makes GetPetStatus temp compile

anishprasad01 committed:
Implemented Status in PetController

anishprasad01 committed:
Implemented Status in PetController

anishprasad01 committed:
Implemented Status in PetController

anishprasad01 committed:
Temporary return in PetService for Status

anishprasad01 committed:
Temporary return in PetService for Status

anishprasad01 committed:
Temporary return in PetService for Status

anishprasad01 committed:
Temporary return in PetService for Status

anishprasad01 committed:
Pet Status endpoint works.

Created a new PetStatus type
Added GetPetStatus to PetService
added Pet Status endpoint to PetController

anishprasad01 committed:
Pet Status endpoint works.

Created a new PetStatus type
Added GetPetStatus to PetService
added Pet Status endpoint to PetController

anishprasad01 committed:
Pet Status endpoint works.

Created a new PetStatus type
Added GetPetStatus to PetService
added Pet Status endpoint to PetController

anishprasad01 committed:
Pet Status endpoint works.

Created a new PetStatus type
Added GetPetStatus to PetService
added Pet Status endpoint to PetController

anishprasad01 committed:
Fixed CodeFactor error

anishprasad01 committed:
Fixed CodeFactor error

anishprasad01 committed:
Fixed CodeFactor error

anishprasad01 committed:
Fixed CodeFactor error

anishprasad01 committed:
Implement Status endpoint for user

anishprasad01 committed:
Implement Status endpoint for user

anishprasad01 committed:
Implement Status endpoint for user

anishprasad01 committed:
Implement Status endpoint for user

anishprasad01 committed:
added no-pet check

added code to return the statuses as Json in UserController

anishprasad01 committed:
added no-pet check

added code to return the statuses as Json in UserController

anishprasad01 committed:
added no-pet check

added code to return the statuses as Json in UserController

anishprasad01 committed:
added no-pet check

added code to return the statuses as Json in UserController

anishprasad01 committed:
Changed all numbers used in GetPetStatus to constants

Changed # of hours in query to get activities to a parameter and constant

Removed StatusRequest class and changed the function signature in PetController

Misc. Code Cleanup

anishprasad01 committed:
Changed all numbers used in GetPetStatus to constants

Changed # of hours in query to get activities to a parameter and constant

Removed StatusRequest class and changed the function signature in PetController

Misc. Code Cleanup

anishprasad01 committed:
Changed all numbers used in GetPetStatus to constants

Changed # of hours in query to get activities to a parameter and constant

Removed StatusRequest class and changed the function signature in PetController

Misc. Code Cleanup

anishprasad01 committed:
Changed all numbers used in GetPetStatus to constants

Changed # of hours in query to get activities to a parameter and constant

Removed StatusRequest class and changed the function signature in PetController

Misc. Code Cleanup

anishprasad01 committed:
Moved position of constants on PetService to top of the class

Fixed an error in UserController caused by removal of StatusRequest

anishprasad01 committed:
Moved position of constants on PetService to top of the class

Fixed an error in UserController caused by removal of StatusRequest

anishprasad01 committed:
Moved position of constants on PetService to top of the class

Fixed an error in UserController caused by removal of StatusRequest

anishprasad01 committed:
Moved position of constants on PetService to top of the class

Fixed an error in UserController caused by removal of StatusRequest

anishprasad01 committed:
Modified the Constant for Hours in PetService to remove the - sign from the query

Added missing @ symbols in query parameter assignment code

anishprasad01 committed:
Modified the Constant for Hours in PetService to remove the - sign from the query

Added missing @ symbols in query parameter assignment code

anishprasad01 committed:
Modified the Constant for Hours in PetService to remove the - sign from the query

Added missing @ symbols in query parameter assignment code

SaamIAm committed:
added basic spam race feature

SaamIAm committed:
added basic spam race feature

SaamIAm committed:
minor formating

SaamIAm committed:
minor formating

anishprasad01 committed:
Added a check in case there are no activities in the interval of hours specified

anishprasad01 committed:
Added a check in case there are no activities in the interval of hours specified

anishprasad01 committed:
Added a check in case there are no activities in the interval of hours specified

anishprasad01 committed:
Fixed negative happiness bug

Added checks for negative hunger and negative happiness

Removed hunger value from happiness calculation

anishprasad01 committed:
Fixed negative happiness bug

Added checks for negative hunger and negative happiness

Removed hunger value from happiness calculation

anishprasad01 committed:
Fixed negative happiness bug

Added checks for negative hunger and negative happiness

Removed hunger value from happiness calculation

anishprasad01 committed:
Merge branch 'implement-status-api' of https://github.com/Chris-Johnston/PGTRTKD3000DGOTY into implement-status-api

anishprasad01 committed:
?

anishprasad01 committed:
?

anishprasad01 committed:
Removed Activity query for Status and replaced it with a call to ActivityService

anishprasad01 committed:
Removed Activity query for Status and replaced it with a call to ActivityService

anishprasad01 committed:
Merge branch 'implement-status-api' of https://github.com/Chris-Johnston/PGTRTKD3000DGOTY into implement-status-api

anishprasad01 committed:
Changing the conditions for Hunger level decrease and increase

anishprasad01 committed:
Modified Hunger decrease code to ensure that hunger will never be > 1 or < 0

anishprasad01 committed:
Modified Status to use a DateTime to specify the number of days to check. Added const variables for number of days and number of activities.

Added Feeding TIme, Race Time, and Training Time to Pet Status type

anishprasad01 committed:
Added check for invalid Happiness percentages.

Fixed Happiness always increasing bug.

Fixed non-nullable activitytype in GetActivity function

anishprasad01 committed:
Fixed issue where Hunger never decreased

Added more comments

Renamed Get pet status fucntion to be more descriptive

anishprasad01 committed:
Merge branch 'master' into implement-status-api

anishprasad01 committed:
Moved functions of Pet Status into helper methods

SaamIAm committed:
added race timer

SaamIAm committed:
added race timer

SaamIAm committed:
added score system

SaamIAm committed:
added score system

Chris-Johnston committed:
Add the pet status detail page

Adds the page that shows the details about a pet status, and including all of the buttons used
to interact with the pet.

Adds 'dead.jpg' image, shown when a pet is dead.

Chris-Johnston committed:
Add the pet status detail page

Adds the page that shows the details about a pet status, and including all of the buttons used
to interact with the pet.

Adds 'dead.jpg' image, shown when a pet is dead.

Chris-Johnston committed:
lint: extra spaces

Chris-Johnston committed:
lint: extra spaces

Chris-Johnston commented:
@SaamIAm resolve merge conflicts

Chris-Johnston commented on PetGame/Pages/Race.cshtml:
```diff
@@ -0,0 +1,123 @@
+    Race Page
+</h1>
+<h2>
+    Spam button race!
```
Weird wording.

anishprasad01 committed:
Moved more functionality into a separate function

Added comments to const variables

Chris-Johnston commented on PetGame/Pages/Race.cshtml:
```diff
@@ -0,0 +1,123 @@
+
+<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+<style>
+    canvas {
```
We should not use inline stylesheets, we already have a stylesheet.

Chris-Johnston commented on PetGame/Pages/Race.cshtml:
```diff
@@ -0,0 +1,123 @@
+}
+
+var myGameArea = {
+    canvas : document.createElement("canvas"),
```
We should avoid creating elements, and instead have them already contained in the html body on load.

Chris-Johnston commented on PetGame/Pages/Race.cshtml:
```diff
@@ -0,0 +1,123 @@
+    myGamePiece.speedX += (distance);
+
+}</script>
+    <div style="text-align:left;width:480px;">
```
again, we should set this style in the css, and not inline

anishprasad01 committed:
Fixed CodeFactor Error

SaamIAm committed:
added reset race button

SaamIAm committed:
added reset race button

Chris-Johnston commented on PetGame/Pages/Race.cshtml:
```diff
@@ -0,0 +1,123 @@
+
+}</script>
+    <div style="text-align:left;width:480px;">
+        <button onclick="moveRight();startCount()">Press Me</button>
```
we should give this button an id, instead of writing js in the `onclick` handler.

SaamIAm replied:
I gave it an Id but am still calling the functions onclick

Chris-Johnston reviewed (changes requested):
Some changes need to be made, but we do need to merge this in eventually so that it can be used.

Chris-Johnston committed:
lint: extra spaces

Chris-Johnston committed:
lint: extra spaces

Chris-Johnston committed:
Merge branch 'initial-pet-status-page' of github.com:Chris-Johnston/PGTRTKD3000DGOTY into initial-pet-status-page

SaamIAm committed:
race time fix

SaamIAm committed:
race time fix

Chris-Johnston committed:
add snarky comment

Chris-Johnston committed:
add logos under img

Chris-Johnston committed:
Add favicons

anishprasad01 committed:
Added comments to Get (status) in UserController

SaamIAm committed:
fixed wording

SaamIAm committed:
fixed wording

Chris-Johnston committed:
Replace the dead image

Chris-Johnston committed:
Add the logo to the landing page, and the layout nav bar

Chris-Johnston committed:
fix the behavior of the hamburger button (mmmm.... hamburger button)

SaamIAm committed:
Merge branch 'master' into add-race-page

anishprasad01 committed:
Status returns 401 for a user who isn't logged in

Changed the use or chars to determine operation to an enum

Changed time checks to TimeSpans

Misc. Code cleanup

anishprasad01 committed:
Further code cleanup

Added more comments

anishprasad01 committed:
Removed extra number. Replaced it with previously declared constant

Chris-Johnston committed:
Merge pull request #65 from Chris-Johnston/add-logos

Add Favicons and Logo Files

SaamIAm committed:
added endurance button cooldown

SaamIAm committed:
added endurance button cooldown

anishprasad01 committed:
Removed ModifyPercentages function to simplify logic

Made cooldown length into a const variable

anishprasad01 committed:
Fixed Error where hunger would become negative if no activities had occured within the time span

anishprasad01 committed:
Fixed code issues in GetPetStatusList

Returns empty list instead of null

Chris-Johnston committed:
use spans instead of divs, make it so that it will be easier to set these values in js later

Chris-Johnston committed:
Merge branch 'master' into initial-pet-status-page

anishprasad01 committed:
Removed PercentageOptions enum

anishprasad01 committed:
Added the missing PostPet function to the UserController

SaamIAm committed:
adjusted strength and endurance

SaamIAm committed:
adjusted strength and endurance

SaamIAm committed:
removed canvas width magic number

SaamIAm committed:
removed canvas width magic number

Chris-Johnston committed:
Merge pull request #57 from Chris-Johnston/implement-status-api

Implement Status API Endpoint in Pet and User

SaamIAm commented:
@Chris-Johnston How many of those comments above are mission critical?

Chris-Johnston committed:
Fix HTTP Unauthorized StatusCode

Chris-Johnston commented:
I would greatly prefer not having to go back to fix them later.

Chris-Johnston committed:
Merge pull request #64 from Chris-Johnston/initial-pet-status-page

Add the (initial) pet status detail page

anishprasad01 committed:
Added services needed to Race codebehind

anishprasad01 committed:
Added services needed to Race codebehind

anishprasad01 committed:
Get Current User and Pet

anishprasad01 committed:
Get Current User and Pet

anishprasad01 committed:
Merge branch 'add-race-page' of https://github.com/Chris-Johnston/PGTRTKD3000DGOTY into add-race-page

anishprasad01 committed:
Rebase onto master

Race Page gets the Current USer and Current Pet

anishprasad01 committed:
Created function to add Race to the DB

anishprasad01 committed:
Removed non-working code
